src/api/repository/controller.go

```diff
@@ -17,38 +17,42 @@ package repository
 import (
 	"context"
 	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils"
 	ierror "github.com/goharbor/harbor/src/internal/error"
+	"github.com/goharbor/harbor/src/pkg/project"
 	"github.com/goharbor/harbor/src/pkg/q"
 	"github.com/goharbor/harbor/src/pkg/repository"
 )
 
 var (
 	// Ctl is a global repository controller instance
-	Ctl = NewController(repository.Mgr)
+	Ctl = NewController()
 )
 
 // Controller defines the operations related with repositories
 type Controller interface {
-	// Ensure the repository specified by the "name" exists under the project,
-	// creates it if it doesn't exist. The "name" should contain the namespace part.
-	// The "created" will be set as true when the repository is created
-	Ensure(ctx context.Context, projectID int64, name string) (created bool, id int64, err error)
+	// Ensure the repository specified by the "name" exists, creates it if it doesn't exist.
+	// The "name" should contain the namespace part. The "created" will be set as true
+	// when the repository is created
+	Ensure(ctx context.Context, name string) (created bool, id int64, err error)
```

Comment:
Why do we need to know if it is a newly created?

Reply:
As the `Ensure` may create the artifact or not, return the `created` can distinguish such cases, but it isn't used yet in our use cases

Reply:
It is there already and isn't introduced by this PR

Reply:
What I mean is it seems we do NOT need to distinguish newly created and existing ones. That  parameter is unnecessary.